### DISCUSSION TITLE
Streaming Factor CPU vs GPU

### DISCUSSION THREAD

**Discussion by alexmoini** [juanmc2005/diart#127] [Q&A]
Does anyone have any insight on the audio speed to segmentation output speed ratio on cpus and/or gpus? I'm sure its model dependent but would be curious what the base model being used performs like.

**Comment by juanmc2005** [comment]
Hi @alexmoini,

As stated in [the paper](https://github.com/juanmc2005/StreamingSpeakerDiarization/blob/main/paper.pdf) (Section 4, paragraph "real time"), it takes 165ms to process a single input with a CPU Intel Cascade Lake 6248, and 50ms with a GPU Nvidia Tesla V100 SXM2, which gives real time factors of 0.33 and 0.1 respectively.

I haven't had the time to measure this on other hardware but it's fairly easy to do with `diart.stream`, which prints the average time taken to process each audio buffer (then divide by the step: 500ms by default).
You can also set the `--cpu` flag to ignore any available GPU.

I would be very interested in reporting the real time factors in a table somewhere in the existing documentation.
If somebody has done any measurements or is willing to help, we could start adding this information.
